solvers/SimpleSOL/ParticleSystems/neutral_particles.hpp

```diff
@@ -239,70 +247,84 @@ class NeutralParticleSystem {
+
       std::vector<double> line_end = {mid_point_x,
                                       this->periodic_bc->global_origin[1] +
                                           this->periodic_bc->global_extent[1]};
 
-      auto tmp_init = std::make_shared<ParticleInitialisationLine>(
-          this->domain, this->sycl_target, line_start, line_end,
-          this->source_line_bin_count);
-      this->source_lines.push_back(tmp_init);
-      this->source_samplers.push_back(
-          std::make_shared<
-              SimpleUniformPointSampler<ParticleInitialisationLine>>(
-              this->sycl_target, tmp_init));
-
-    } else if (this->source_line_count == 2) {
+      region_lines.push_back(std::make_pair(line_start, line_end));
+    } else if (this->source_region_count == 2) {
 
       const double lower_x =
           this->periodic_bc->global_origin[0] +
-          this->source_line_offset * this->periodic_bc->global_extent[0];
+          this->source_region_offset * this->periodic_bc->global_extent[0];
       const double upper_x = this->periodic_bc->global_origin[0] +
-                             (1.0 - this->source_line_offset) *
+                             (1.0 - this->source_region_offset) *
                                  this->periodic_bc->global_extent[0];
-
       // lower line
       std::vector<double> line_start0 = {lower_x,
                                          this->periodic_bc->global_origin[1]};
       std::vector<double> line_end0 = {lower_x,
                                        this->periodic_bc->global_origin[1] +
                                            this->periodic_bc->global_extent[1]};
-
-      auto tmp_init0 = std::make_shared<ParticleInitialisationLine>(
-          this->domain, this->sycl_target, line_start0, line_end0,
-          this->source_line_bin_count);
-      this->source_lines.push_back(tmp_init0);
-      this->source_samplers.push_back(
-          std::make_shared<
-              SimpleUniformPointSampler<ParticleInitialisationLine>>(
-              this->sycl_target, tmp_init0));
-
+      region_lines.push_back(std::make_pair(line_start0, line_end0));
       // upper line
       std::vector<double> line_start1 = {upper_x,
                                          this->periodic_bc->global_origin[1]};
       std::vector<double> line_end1 = {upper_x,
                                        this->periodic_bc->global_origin[1] +
                                            this->periodic_bc->global_extent[1]};
 
-      auto tmp_init1 = std::make_shared<ParticleInitialisationLine>(
-          this->domain, this->sycl_target, line_start1, line_end1,
-          this->source_line_bin_count);
-      this->source_lines.push_back(tmp_init1);
-      this->source_samplers.push_back(
-          std::make_shared<
-              SimpleUniformPointSampler<ParticleInitialisationLine>>(
-              this->sycl_target, tmp_init1));
-
```

Comment:
I think this logic will need to account for the fact that periodic_bc->global_extent - periodic_bc->global_origin is no longer the size of the mesh once it's rotated.

Reply:
The rotation is done per line on new line 316 and 317 by the `rotate` lambda. Does that solve this problem? Can we run it and check?

Reply:
I don't think it does -
Effectively I think we want to set up the region centre line(s) in the 'problem frame', spawn the offset lines for each region, then rotate them into the x-y frame.
At the moment we're setting up the centre lines in the x-y frame, because that's how periodic_bc->global_origin[*] and this->periodic_bc->global_extent[*] are set up.

Reply:
Right now, we're using the length data (global_origin/_extent) to set up the lines and then rotating them. Do we need to set them up according to a different length that has been pre-rotated the other way?